jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/json/SpanDeserializer.java

```diff
@@ -43,26 +46,38 @@ public Span deserialize(JsonParser jp, DeserializationContext ctxt) throws IOExc
     JsonNode node = objectMapper.getFactory().setCodec(objectMapper).getCodec().readTree(jp);
 
     String spanIdHex = node.get("spanID").asText();
-    String parentIdHex = node.get("parentSpanID").asText();
     String traceIdHex = node.get("traceID").asText();
     String startTimeStr = node.get("startTime").asText();
 
-    BigInteger spanId = new BigInteger(spanIdHex, 16);
-    BigInteger parentId = new BigInteger(parentIdHex, 16);
-
     JsonNode processNode = node.get("process");
     Process process = objectMapper.treeToValue(processNode, Process.class);
 
     JsonNode tagsNode = node.get("tags");
     KeyValue[] tags = objectMapper.treeToValue(tagsNode, KeyValue[].class);
 
     Span span = new Span();
-    span.setSpanId(spanId.longValue());
-    span.setParentId(parentId.longValue());
+    span.setSpanId(new BigInteger(spanIdHex, 16).longValue());
     span.setTraceId(traceIdHex);
+    span.setRefs(deserializeReferences(node));
     span.setStartTime(startTimeStr != null ? Long.parseLong(startTimeStr) : null);
     span.setProcess(process);
     span.setTags(Arrays.asList(tags));
     return span;
   }
+
+  private List<Reference> deserializeReferences(JsonNode node) throws JsonProcessingException {
```

Comment:
Would it be worth creating a test for this?

Reply:
We don't have any serialization/deserialization tests in this repo. Just directly running integration tests with real j-collector and ES.

I would skip it not, just for simplicity. I think you have noticed but I have introduced stages so CI runs with multiple jaeger versions.

Reply:
Actually there is one deserialization test. This would be more complicated as passed JNode is for span not only Reference